### PR TITLE
Add numpy.fft.fftn frontend 

### LINKED ISSUE #17028: fftn

### PR DESCRIPTION
**Pull request by RohitSgh**
Close #17028

### CODE REVIEW

**Comment by RohitSgh** [comment]
add_frontend_checklist

**Comment by RohitSgh** [comment]
Hi @sherry30. Can you please help me why checks are failing?

**Comment by RohitSgh** [comment]
add_frontend_checklist

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by RohitSgh** [comment]
add_frontend_checklist

**Comment by RohitSgh** [comment]
Hi @sherry30. Can you please help me why checks are failing?

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

You can ignore the `key error: "cpu"` error since that is unrelated  to your changes.
After running it on an older version, I get 
```
>       s = list(s)
E       TypeError: 'int' object is not iterable
```

This is because `s` is being passed as an integer while numpy expects a sequence of ints. You should generate what is expected by the framework function.

thanks

**Comment by RohitSgh** [comment]
add_frontend_checklist

**Comment by RohitSgh** [comment]
Please check again @sherry30 

Can more derails be provide on what exactly `intelligent-tests-pr / run_tests (1) (pull_request)` and `intelligent-tests-pr / run_tests (24) (pull_request)` are?

**Comment by RohitSgh** [comment]
Also, my issue is not being added to list https://github.com/unifyai/ivy/issues/1532. Have tried both way `#17028` and `- [ ] #17028`

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_fft/test_discrete_fourier_transform.py [hunk 1]
```diff
@@ -192,3 +192,32 @@ def test_numpy_rfftfreq(n, sample_rate, frontend, test_flags, fn_tree, on_device
...
+@handle_frontend_test(
+    fn_tree="numpy.fft.fftn",
+    dtype_input_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("float_and_complex"),
+        shape=(2, 2),
+        min_axis=-2,
+        force_int_axis=True,
+    ),
+    norm=st.sampled_from(["backward", "ortho", "forward"]),
+    s=st.integers(min_value=2, max_value=10),
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_fft/test_discrete_fourier_transform.py:206]: `s` is being passed as an integer here, you can use `helpers.lists` if you want to generate a sequence of inetgers

**Review by sherry30 — CHANGES_REQUESTED** [review]
the integer problem hasn't been resolved, which is why its failing for me

**Comment by sherry30** [comment]
> Also, my issue is not being added to list #1532. Have tried both way `#17028` and `- [ ] #17028`

I've added your issue manually in the list

**Comment by RohitSgh** [comment]
add_frontend_checklist

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_fft/test_discrete_fourier_transform.py (hunk 1 above)
Comment (RohitSgh) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_fft/test_discrete_fourier_transform.py]: Please check now, @sherry30

**Comment by RohitSgh** [comment]
Please review @sherry30. Only 1 failing check now `intelligent-tests-pr / run_tests (24) (pull_request)`

**Review by sherry30 — CHANGES_REQUESTED** [review]
Minor change requested

**Comment by RohitSgh** [comment]
> Please review @sherry30. Only 1 failing check now `intelligent-tests-pr / run_tests (24) (pull_request)`

Now, also 1 failing, but is has changed to `intelligent-tests-pr / run_tests (1) (pull_request)`, @sherry30

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

You can actually click on the failing tests and see the test failures.
It would be best if you can go through the failure and try to resolve it yourself. You can ask me if you get stuck anywhere. I'd be happy to help in that case.

Thanks

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by RohitSgh** [comment]
Working @ivy-seed

**Pull request closed without merging**